lib/reply.js

```diff
@@ -138,9 +138,24 @@ Reply.prototype.send = function (payload) {
 
   if (this[kReplySerializer] !== null) {
     payload = this[kReplySerializer](payload)
-  } else if (hasContentType === false || contentType.indexOf('application/json') > -1) {
-    if (hasContentType === false || contentType.indexOf('charset') === -1) {
+
+  // We test below if we have a custom JSON type (e.g. 'application/hal+json' or 'application/ld+json')
+  } else if (hasContentType === false || /^application\/([a-z.-]+\+)?json/.test(contentType)) {
+    if (hasContentType === false) {
       this[kReplyHeaders]['content-type'] = CONTENT_TYPE.JSON
+    } else {
+      // If hasContentType === true, we have a JSON mimetype
+      if (contentType.indexOf('charset') === -1) {
```

Comment:
this is lots of logic that most users are unlikely to need, but that is going to slow them down. wonder if it can be made opt-in or pluginized somehow.

Reply:
This verification was done before. I just moved it to an else. It is true that we have more logic than before. However, I think pluginize it will add unnecessary overhead somewhere else (e.g. need for a configuration or new setting). I haven't had ideas about how to do it. The new commit improves it by having just one extra verification for the base case.